api/v1alpha3/vspheremachinetemplate_webhook.go

```diff
@@ -17,11 +17,61 @@ limitations under the License.
 package v1alpha3
 
 import (
+	"errors"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
 func (r *VSphereMachineTemplate) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(r).
 		Complete()
 }
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-vspheremachinetemplate,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=vspheremachinetemplates,versions=v1alpha3,name=validation.vspheremachinetemplate.infrastructure.x-k8s.io,sideEffects=None
+
+var _ webhook.Validator = &VSphereMachineTemplate{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *VSphereMachineTemplate) ValidateCreate() error {
+	var allErrs field.ErrorList
+	spec := r.Spec.Template.Spec
+
+	if spec.Template == "" {
+		allErrs = append(allErrs, field.Forbidden(field.NewPath("spec", "template", "spec", "template"), "cannot be empty"))
```

Comment:
```suggestion
		allErrs = append(allErrs, field.Invalid(field.NewPath("spec", "template", "spec", "template"), "cannot be empty"))
```
Or possibly, field.Required